src/controllers/payment.callback.controller.ts

```diff
@@ -1,12 +1,23 @@
 import { NextFunction, Request, Response } from "express";
 import { urlUtils } from "../utils/url";
-import { logger } from "../utils/logger";
+import { createAndLogError, logger } from "../utils/logger";
 import { CONFIRMATION_PATH, REVIEW_PATH } from "../types/page.urls";
+import { Session } from "@companieshouse/node-session-handler";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
   try {
+    const session = req.session as Session;
     const paymentStatus = req.query.status;
     const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const returnedState = req.query.state;
+    const state = session.getExtraData("payment-nonce");
```

Comment:
could you rename the state const to savedState? think it might be clearer that way